src/category_theory/sites/grothendieck.lean

```diff
@@ -0,0 +1,277 @@
+  rw [sieve.pullback_inter, R.pullback_eq_top_of_mem Hf],
+  simp [sj],
+end
+
+@[simp]
+lemma intersection_covering_iff : R ⊓ S ∈ J X ↔ R ∈ J X ∧ S ∈ J X :=
+⟨λ h, ⟨J.superset_covering inf_le_left h, J.superset_covering inf_le_right h⟩,
+ λ t, intersection_covering _ t.1 t.2⟩
+
+/--
+The sieve `S` on `X` `J`-covers an arrow `f` to `X` if `S.pullback f ∈ J Y`.
+This definition is an alternate way of presenting a Grothendieck topology.
+-/
+def covers (S : sieve X) (f : Y ⟶ X) : Prop := S.pullback f ∈ J Y
+
+lemma covers_iff (S : sieve X) (f : Y ⟶ X) : J.covers S f ↔ S.pullback f ∈ J Y :=
+iff.rfl
+
+lemma covering_iff_covers_id (S : sieve X) : S ∈ J X ↔ J.covers S (𝟙 X) :=
+by simp [covers_iff]
+
+/-- The maximality axiom in 'arrow' form: Any arrow `f` in `S` is covered by `S`. -/
+lemma arrow_max (f : Y ⟶ X) (S : sieve X) (hf : S.arrows f) : J.covers S f :=
+begin
+  rw [covers, (sieve.pullback_eq_top_iff_mem f).1 hf],
+  apply J.top_mem,
+end
+
+/-- The stability axiom in 'arrow' form: If `S` covers `f` then `S` covers `g ≫ f` for any `g`. -/
+lemma arrow_stab (f : Y ⟶ X) (S : sieve X) (h : J.covers S f) {Z : C} (g : Z ⟶ Y) :
+  J.covers S (g ≫ f) :=
+begin
+  rw covers_iff at h ⊢,
+  simp [h, sieve.pullback_comp],
+end
+
+/--
+The transitivity axiom in 'arrow' form: If `S` covers `f` and every arrow in `S` is covered by
+`R`, then `R` covers `f`.
+-/
+lemma arrow_trans (f : Y ⟶ X) (S R : sieve X) (h : J.covers S f) :
+  (∀ {Z : C} (g : Z ⟶ X), S.arrows g → J.covers R g) → J.covers R f :=
+begin
+  intro k,
+  apply J.transitive h,
+  intros Z g hg,
+  rw ← sieve.pullback_comp,
+  apply k (g ≫ f) hg,
+end
+
+lemma arrow_intersect (f : Y ⟶ X) (S R : sieve X) (hS : J.covers S f) (hR : J.covers R f) :
+  J.covers (S ⊓ R) f :=
+by simpa [covers_iff] using and.intro hS hR
+
+instance : partial_order (grothendieck_topology C) :=
+{ le := λ J₁ J₂, (J₁ : Π (X : C), set (sieve X)) ≤ (J₂ : Π (X : C), set (sieve X)),
+  le_refl := λ J₁, le_refl _,
+  le_trans := λ J₁ J₂ J₃ h₁₂ h₂₃, le_trans h₁₂ h₂₃,
+  le_antisymm := λ J₁ J₂ h₁₂ h₂₁, grothendieck_topology.ext _ _ (le_antisymm h₁₂ h₂₁) }
+
+instance : has_Inf (grothendieck_topology C) :=
+{ Inf := λ T,
+  { sieves := Inf (sieves '' T),
+    top_mem' :=
+    begin
+      rintro X S ⟨⟨_, J, hJ, rfl⟩, rfl⟩,
+      simp,
+    end,
+    pullback_stable' :=
+    begin
+      rintro X Y S hS f _ ⟨⟨_, J, hJ, rfl⟩, rfl⟩,
+      apply J.pullback_stable _ (f _ ⟨⟨_, _, hJ, rfl⟩, rfl⟩),
+    end,
+    transitive' :=
+    begin
+      rintro X S hS R h _ ⟨⟨_, J, hJ, rfl⟩, rfl⟩,
+      apply J.transitive (hS _ ⟨⟨_, _, hJ, rfl⟩, rfl⟩) _ (λ Y f hf, h hf _ ⟨⟨_, _, hJ, rfl⟩, rfl⟩),
+    end } }
+
+instance : complete_lattice (grothendieck_topology C) :=
+complete_lattice_of_Inf _
+begin
+  intro T,
+  refine @is_glb.of_image _ _ _ _ sieves _ _ _ _,
+  { intros, refl },
+  { exact is_glb_Inf _ },
+end
+
+variable (C)
+/-- The trivial Grothendieck topology, in which only the maximal sieve is covering. -/
```

Comment:
You already know that there is a complete lattice structure. So there is already a `\bot` and `\top`.
Could you make these the bot and top definitionally?

Reply:
I've amended, is this the sort of thing you had in mind?

Reply:
yup, looks good